pkg/ipamd/ipamd_test.go

```diff
@@ -1417,26 +1415,29 @@ func TestIPAMContext_filterUnmanagedENIs(t *testing.T) {
 		eni2.ENIID: {"hi": "tag", eniNodeTagKey: "i-abcdabcdabcd"},
 		eni3.ENIID: {"hi": "tag", eniNodeTagKey: instanceID}}
 
-	mockAWSUtils := mock_awsutils.NewMockAPIs(ctrl)
-	mockAWSUtils.EXPECT().GetPrimaryENI().Times(5).Return(eni1.ENIID)
-	mockAWSUtils.EXPECT().GetInstanceID().Times(3).Return(instanceID)
-
 	tests := []struct {
-		name          string
-		tagMap        map[string]awsutils.TagMap
-		enis          []awsutils.ENIMetadata
-		want          []awsutils.ENIMetadata
-		unmanagedenis []string
+		name                       string
+		tagMap                     map[string]awsutils.TagMap
+		enis                       []awsutils.ENIMetadata
+		want                       []awsutils.ENIMetadata
+		unmanagedenis              []string
+		expectedGetPrimaryENICalls int
+		expectedGetInstanceIDCalls int
 	}{
-		{"No tags at all", nil, allENIs, allENIs, nil},
-		{"Primary ENI unmanaged", Test1TagMap, allENIs, allENIs, nil},
-		{"Secondary/Tertiary ENI unmanaged", Test2TagMap, allENIs, primaryENIonly, []string{eni2.ENIID, eni3.ENIID}},
-		{"Secondary ENI unmanaged", Test3TagMap, allENIs, filteredENIonly, []string{eni2.ENIID}},
-		{"Secondary ENI unmanaged and Tertiary ENI CNI created", Test4TagMap, allENIs, filteredENIonly, []string{eni2.ENIID}},
-		{"Secondary ENI not CNI created and Tertiary ENI CNI created", Test5TagMap, allENIs, filteredENIonly, nil},
+		{"No tags at all", nil, allENIs, allENIs, nil, 0, 0},
```

Comment:
It's funny. I realized that the test fix I had to do was the same as described in this blogpost - https://go.dev/blog/loopvar-preview